src/scheduler/pbs_sched.cpp

```diff
@@ -530,204 +506,139 @@ close_server(sched_svrconn *sconn)
+
+
+	while (clust_primary_sock < 0 || clust_secondary_sock < 0 || registered == 0) {
+		/* pbs_connect() will return a connection handle for all servers */
+		if (clust_primary_sock < 0) {
+			clust_primary_sock = pbs_connect(NULL);
+			if (clust_primary_sock < 0) {
 				/* wait for 2s for not to burn too much CPU, and then retry connection */
 				sleep(2);
 				continue;
 			}
 		}
-		secondary_sock = connect_server_helper(svrhost, 0);
-		if (secondary_sock < 0) {
+		clust_secondary_sock = pbs_connect(NULL);
+		if (clust_secondary_sock < 0) {
 			/* wait for 2s for not to burn too much CPU, and then retry connection */
 			sleep(2);
 			continue;
 		}
-	}
-	if (tpp_em_add_fd(poll_context, secondary_sock, EM_IN | EM_HUP | EM_ERR) < 0) {
-		log_errf(errno, __func__, "Couldn't add secondary connection to poll list for server %s", svrhost);
-		pbs_disconnect(primary_sock);
-		pbs_disconnect(secondary_sock);
-		die(-1);
-	}
 
-	svr = static_cast<sched_svrconn *>(malloc(sizeof(sched_svrconn)));
-	if (!svr) {
-		log_err(errno, __func__, MEM_ERR_MSG);
-		die(-1);
-	}
+		svr_conns_primary =  static_cast<svr_conn_t *>(get_conn_svr_instances(clust_primary_sock));
+		svr_conns_secondary =  static_cast<svr_conn_t *>(get_conn_svr_instances(clust_secondary_sock));
 
-	svr->svrhost = strdup(svrhost);
-	if (!svr->svrhost) {
-		free(svr);
-		log_err(errno, __func__, MEM_ERR_MSG);
-		die(-1);
+		for (i = 0; i < num_conf_svrs; i++) {
+			if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN || svr_conns_primary[i].state == SVR_CONN_STATE_DOWN) {	
+				if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN)
+					clust_primary_sock = -1;
+				if (svr_conns_primary[i].state == SVR_CONN_STATE_DOWN)
+					clust_secondary_sock = -1;
+				break;
+			}
+		}
+
+		if (i != num_conf_svrs) {
+			/* If we reached here means one of the servers is down or not connected
+			 * we should go to the top of the loop again and call pbs_connect
+			 * Also wait for 2s for not to burn too much CPU
+			 */
+			sleep(2);
+			continue;
+		}
+
+		if (pbs_register_sched(sc_name) == 0) {
+			log_errf(pbs_errno, __func__, "Couldn't register the scheduler %s with the configured servers", sc_name);
+			/* wait for 2s for not to burn too much CPU, and then retry connection */
+			sleep(2);			
+			continue;
+		} else
+			registered = 1;
+		
 	}
-	svr->primary_sock = primary_sock;
-	svr->secondary_sock = secondary_sock;
-	svrs_newarr = add_ptr_to_array(servers, svr);
-	if (svrs_newarr == NULL) {
-		close_server(svr);
-		die(-1);
+
+
+	for (i = 0; i < get_num_servers(); i++) {
+		if (tpp_em_add_fd(poll_context, svr_conns_secondary[i].sd, EM_IN | EM_HUP | EM_ERR) < 0) {
+			log_errf(errno, __func__, "Couldn't add secondary connection to poll list for server %s", svr_conns_secondary[i].name);
+			die(-1);
+		}
 	}
-	servers = (sched_svrconn **)svrs_newarr;
-	log_eventf(PBSEVENT_ADMIN | PBSEVENT_FORCE, PBS_EVENTCLASS_SCHED, LOG_INFO, msg_daemonname, "Connected to server %s", svrhost);
```

Comment:
I think this log message can be useful, why did you decide to remove it?

Reply:
Since pbs_connect now connects to all the configured servers I removed it earlier. Added a similar log message which says all the configured servers are connected.